pkg/microservice/aslan/core/code/client/gitee/gitee.go

```diff
@@ -0,0 +1,141 @@
+package gitee
+
+import (
+	"context"
+
+	giteeClient "gitee.com/openeuler/go-gitee/gitee"
+	"github.com/antihax/optional"
+	"go.uber.org/zap"
+
+	"github.com/koderover/zadig/pkg/microservice/aslan/config"
+	"github.com/koderover/zadig/pkg/microservice/aslan/core/code/client"
+	"github.com/koderover/zadig/pkg/tool/gitee"
+)
+
+type Config struct {
+	AccessToken string `json:"access_token"`
+	EnableProxy bool   `json:"enable_proxy"`
+}
+
+type Client struct {
+	Client      *gitee.Client
+	AccessToken string
+}
+
+func (c *Config) Open(id int, logger *zap.SugaredLogger) (client.CodeHostClient, error) {
+	client := gitee.NewClient(id, c.AccessToken, config.ProxyHTTPSAddr(), c.EnableProxy)
+	return &Client{
+		Client:      client,
+		AccessToken: c.AccessToken,
+	}, nil
+}
+
+func (c *Client) ListBranches(opt client.ListOpt) ([]*client.Branch, error) {
+	bList, err := c.Client.ListBranches(context.TODO(), opt.Namespace, opt.ProjectName, nil)
+	if err != nil {
+		return nil, err
+	}
+	var res []*client.Branch
+	for _, o := range bList {
+		res = append(res, &client.Branch{
+			Name:      o.Name,
+			Protected: o.Protected,
+		})
+	}
+	return res, nil
+}
+
+func (c *Client) ListTags(opt client.ListOpt) ([]*client.Tag, error) {
+	tags, err := c.Client.ListTags(context.TODO(), c.AccessToken, opt.Namespace, opt.ProjectName)
+	if err != nil {
+		return nil, err
+	}
+	var resp []*client.Tag
+	for _, tag := range tags {
+		resp = append(resp, &client.Tag{
+			Name:    tag.Name,
+			Message: tag.Message,
+		})
+	}
+
+	return resp, nil
+}
+
+func (c *Client) ListPrs(opt client.ListOpt) ([]*client.PullRequest, error) {
+	prs, err := c.Client.ListPullRequests(context.TODO(), opt.Namespace, opt.ProjectName, &giteeClient.GetV5ReposOwnerRepoPullsOpts{
+		PerPage: optional.NewInt32(100),
```

Comment:
Does it mean only the latest 100 PR's will be returned?

Reply:
Yes, that's how github currently handles it